libvast/include/vast/dump_operator.hpp

```diff
@@ -0,0 +1,46 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/logical_operator.hpp"
+
+#include <vast/plugin.hpp>
+
+namespace vast {
+
+/// The logical operator for dumping data that will have to be joined later
+/// during pipeline execution.
+class dump_operator : public logical_operator<chunk_ptr, void> {
+public:
+  explicit dump_operator(const dumper_plugin& dumper) noexcept
+    : dumper_plugin_{dumper} {
+  }
+
+  [[nodiscard]] auto
+  make_physical_operator(const type& input_schema,
```

Comment:
The signature of this function changed with #3039; I would like to take a second look at this PR after you've rebased it on current main, as I think most of the diff will magically disappear then. The logic for the operator does look good to me, though.

Reply:
Rebased & realigned logic in https://github.com/tenzir/vast/pull/3024/commits/5459167226ee2f0dcf1c633db42818a96f43988d